### DISCUSSION TITLE
Difference in ImageJ local thickness and Porespy local thickness output

### DISCUSSION THREAD

**Discussion by DMaxJ** [PMEAL/porespy#696] [General]
I asked a question here https://github.com/PMEAL/porespy/discussions/662. Now I want to show some results that describe the difference between both tools. The same 3D stack is used and default settings of porespy applied while computing local thickness. In terms of mean and maximum values they are comparable but not the same. While ImageJ produces a more continuous range of diameters, porespy seems to have jumps in the calculated radii values. 

Is there any explanation to this behaviour? @jgostick @ma-sadeghi 
Thanks in advance!!

<img width="280" alt="testlocalthickness" src="https://user-images.githubusercontent.com/24526540/169253862-7ea8544c-df2d-49b6-8b2e-c2f75f0dd900.png">

**Comment by jgostick** [comment]
PoreSpy's ``local_thickness`` filter operations differently than ImageJ's.  I'm not exactly sure how's there's works, but our operates by just inserting spheres of different radius into the image, which can only be of integer sizes.  If you want to make sure that ALL sizes get used, then set the ``sizes`` argument a high values (which takes longer), or pass in a list of ``sizes`` like ``np.arange(1, 100)`` and the function will use those.  

Note when using the porespy version, if the number of ``sizes`` is specified, then the function will generate a list of sizes between the min and mix of the distance transform, which results in floating point values.  This value is put into the image, even though spheres of integer sizes were drawn.

**Comment by DMaxJ** [comment]
Thank you. I have tried with different options and found that the maximum radii is 30px, which is the same as the value from ImageJ. Although the distribution is slightly different (but pretty close to ImageJ) there are no "gaps". The distribution remains the same after certain size (eg np.arange(1, 50) and np.arange(1,100) produces same output). This means that it is a good idea to try with different sizes.
